LDAR_Sim/testing/unit_testing/test_initialization/test_sources/test_generate_emissions.py

```diff
@@ -143,22 +143,22 @@ def simple_inputs():
 
 
 def expected_dictionary():
-    return {"source1": [date(2021, 5, 21)]}
+    return {"source1": [date(2021, 5, 20)]}
```

Comment:
Why are all these dates now 1 day earlier?

Reply:
Turns out when we do range(self._emis_duration, 1, -1): it runs 364 times because it's exclusive. 
So we were missing a single roll for emissions. with the new optimized way, it should be easier to spot these mistakes.